IPython/html/widgets/widget.py

```diff
@@ -391,13 +417,13 @@ def _trait_to_json(self, x):
         Replace any widgets with their model_id
         """
         if isinstance(x, dict):
-            return {k: self._trait_to_json(v) for k, v in x.items()}
+            return {k: self._trait_to_json(v) for k, v in x.items()}, None
         elif isinstance(x, (list, tuple)):
-            return [self._trait_to_json(v) for v in x]
+            return [self._trait_to_json(v) for v in x], None
         elif isinstance(x, Widget):
-            return "IPY_MODEL_" + x.model_id
+            return "IPY_MODEL_" + x.model_id, None
```

Comment:
As the next step, you should try using this new API to remove this and use custom (de)serializers to handle it.